Source/WebCore/page/Navigation.cpp

```diff
@@ -93,12 +93,12 @@ bool Navigation::canGoForward() const
 }
 
 // https://html.spec.whatwg.org/multipage/nav-history-apis.html#getting-the-navigation-api-entry-index
-static std::optional<size_t> getEntryIndexOfHistoryItem(const Vector<Ref<NavigationHistoryEntry>>& entries, const HistoryItem& item)
+static std::optional<size_t> getEntryIndexOfHistoryItem(const Vector<Ref<NavigationHistoryEntry>>& entries, const HistoryItem& item, size_t start = 0)
 {
     size_t index = 0;
     // FIXME: We could have a more efficient solution than iterating through a list.
     for (auto& entry : entries) {
-        if (entry->associatedHistoryItem() == item)
+        if (entry->associatedHistoryItem() == item && index >= start)
```

Comment:
This is starting to diverge from the spec, which I understand we need to regarding cross-document histories, but I'd like to see more comments documenting the differences.

Reply:
Yes, it is unfortunate. I considered doing a reverse traversal, but I don't think that is much better if at all.